src/components/MessageList/hooks/MessageList/useUnreadMessagesNotification.ts

```diff
@@ -54,5 +61,17 @@ export const useUnreadMessagesNotification = ({
     };
   }, [intersectionObserverIsSupported, messages, unreadCount]);
 
+  useEffect(() => {
+    /**
+     * Handle situation when scrollToBottom is called from another component when the msg list is scrolled above the observed target (unread separator).
+     * The intersection observer is not triggered when Element.scrollTo() is called. So we end up in a situation when we are scrolled to the bottom
+     * and at the same time scrolled above the observed target.
+     */
+
+    if (unreadCount && isMessageListScrolledToBottom && isScrolledAboveTargetTop.current) {
+      setShow(true);
```

Comment:
Looks like this can be done safely outside of the effect. Plus, you'll save yourself one commit to DOM, which is always nice :)

Also, should we update `isScrolledAboveTargetTop.current = false` here, or am I misunderstanding the problem?